internal/services/synapse/synapse_workspace_key_resource_test.go

```diff
@@ -15,13 +15,37 @@ import (
 
 type SynapseWorkspaceKeysResource struct{}
 
-func TestAccSynapseWorkspaceKeys_customerManagedKey(t *testing.T) {
+func TestAccSynapseWorkspaceKeys_basic(t *testing.T) {
 	data := acceptance.BuildTestData(t, "azurerm_synapse_workspace_key", "test")
 	r := SynapseWorkspaceKeysResource{}
 
 	data.ResourceTest(t, r, []acceptance.TestStep{
 		{
-			Config: r.customerManagedKey(data),
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		// CMK takes a while to activate, so validation against the plan tends to fail.
```

Comment:
we should be waiting on activation then? with a state wait until it is done so this never fails?

Reply:
Hi @katbyte ,

Yes, I've added the waiting part in the first commit. And I've removed this comments in the second commit. Please take another look, thanks!

Reply:
only the the update function? what about create? shouldn't it be there too?

Reply:
Yes, you're right, I've added another commit to add it in both create and update.